### PR TITLE
Feature/smd1 4

### PR DESCRIPTION
**Pull request by hbgwhite**
This pull request adds the Depth of Knowledge (DOK) and Allow Calculator fields to the item report. It sends an error to the error report if the subject is math and the item is missing the allow calculator element. It also checks all image tags in items and stimuli to ensure that they include alt tags and reports if they do not. 

I wrote unit tests around the logic that determines the existence of a valid alt attribute within an img tag. These unit tests appear in a separate project. Some minor reorganization had to occur to migrate the existing top-level project down into its own folder to make room for its test sibling. The folder structure now includes the .sln in the top level with the main and test projects as siblings.

### CODE REVIEW

File: TabulateSmarterTestContentPackage/Tabulator.cs [hunk 1]
```diff
@@ -2289,6 +2328,17 @@ static string ToDependsOnString(string itemId, string dependsOnId)
...
 
+        internal bool ReportMissingImgAltTags(string input)
+        {
+            const string imgAltMatcherPattern = @"<\w*img[^>]*>";
+            var result = Regex.Match(input, imgAltMatcherPattern);
+            return result.Success
+                && !string.IsNullOrEmpty(result.Value) 
+                && (!result.Value.Contains("alt=")                         // There is an img tag, but not alt tag
+                || result.Value.Contains("alt=\"\"")                       // Empty alt tag with double quotes
+                || result.Value.Contains("alt=\'\'"));                     // Empty alt tag with single quotes
```
Comment (jtreuting) [inline TabulateSmarterTestContentPackage/Tabulator.cs]: I think using a RegEx would be better on the result.Value.  Something like: alt\w*=\w*[\"'](.+?)[\"']

File: TabulateSmarterTestContentPackageTests/ImgTagAltAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            //act/assert
+            RunValidation(element);
+        }
+
+        [Test]
+        public void ValidImgTagWithEmptyAttributeSingleQuoteShouldReturnTrue()
+        {
+            // arrange
+            const string element = "<img id = \"item_1434_graphics1\" src = \"item_1434_v45_graphics1_png256.png\" " +
+                                   "width = \"888\" height = \"865\" style = \"vertical-align:baseline;\" alt=\'\' />";
```
Comment (jtreuting) [inline TabulateSmarterTestContentPackageTests/ImgTagAltAttributeTests.cs]: Don't need the \ for the ', only for doulbe quotes.

File: .gitignore [hunk 3]
```diff
@@ -98,6 +98,8 @@ publish/
 # NuGet Packages Directory
 ## TODO: If you have NuGet Package Restore enabled, uncomment the next line
 #packages/
+#/TabulateSmarterTestContentPackage/packages/
```
Comment (hbgwhite) [inline .gitignore]: Professional 2015 - The project was set up with the .sln file in the same directory as the project file, so I modified the .sln paths and moved the project file underneath the .sln so I could add the test project as a sibling.

File: TabulateSmarterTestContentPackage/Tabulator.cs (hunk 1 above)
Comment (hbgwhite) [inline TabulateSmarterTestContentPackage/Tabulator.cs]: @"alt(\s*\t*)=(\s*\t*)['""]\w+['""]" is what i ended up with (passed my unit tests).

File: TabulateSmarterTestContentPackageTests/ImgTagAltAttributeTests.cs (hunk 2 above)
Comment (hbgwhite) [inline TabulateSmarterTestContentPackageTests/ImgTagAltAttributeTests.cs]: Removed

File: TabulateSmarterTestContentPackage/Tabulator.cs [hunk 4]
```diff
@@ -243,8 +241,8 @@ private void Initialize(string reportPrefix)
             mErrorReportPath = string.Concat(reportPrefix, cErrorReportFn);
             if (File.Exists(mErrorReportPath)) File.Delete(mErrorReportPath);
 
-            mItemReport = new StreamWriter(string.Concat(reportPrefix, cItemReportFn), false, Encoding.UTF8);
-            mItemReport.WriteLine("Folder,ItemId,ItemType,Version,Subject,Grade,Rubric,AsmtType,Standard,Claim,Target,WordlistId,ASL,BrailleType,Translation,Media,Size");
+            mItemReport = new StreamWriter(string.Concat(reportPrefix, cItemReportFn), false, Encoding.UTF8); // DOK is "Depth of Knowledge"
+            mItemReport.WriteLine("Folder,ItemId,ItemType,Version,Subject,Grade,Rubric,AsmtType,Standard,Claim,Target,WordlistId,ASL,BrailleType,Translation,Media,Size,DOK,AllowCalculator");
```
Comment (hbgwhite) [inline TabulateSmarterTestContentPackage/Tabulator.cs]: Moved

File: TabulateSmarterTestContentPackage.sln [hunk 5]
```diff
@@ -3,7 +3,9 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Express 2013 for Windows Desktop
 VisualStudioVersion = 12.0.30110.0
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TabulateSmarterTestContentPackage", "TabulateSmarterTestContentPackage.csproj", "{9B4A55D6-DB06-48F7-8B61-0F4017E37D0B}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TabulateSmarterTestContentPackage", ".\TabulateSmarterTestContentPackage\TabulateSmarterTestContentPackage.csproj", "{9B4A55D6-DB06-48F7-8B61-0F4017E37D0B}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TabulateSmarterTestContentPackageTests", ".\TabulateSmarterTestContentPackageTests\TabulateSmarterTestContentPackageTests.csproj", "{B7C95605-8322-4AA6-BD38-7676C4C93825}"
```
Comment (hbgwhite) [inline TabulateSmarterTestContentPackage.sln]: Done

**Pull request closed without merging**